ballerina/executor_visitor.bal

```diff
@@ -83,23 +83,15 @@ class ExecutorVisitor {
     public isolated function visitVariable(parser:VariableNode variableNode, anydata data = ()) {}
 
     isolated function executeQuery(parser:FieldNode fieldNode, parser:RootOperationType operationType) {
-        (string|int)[] path = [fieldNode.getName()];
-        string operationTypeName = getOperationTypeNameFromOperationType(operationType);
-        __Type parentType = <__Type>getTypeFromTypeArray(self.schema.types, operationTypeName);
-        __Type fieldType = getFieldTypeFromParentType(parentType, self.schema.types, fieldNode);
-        Field 'field = new (fieldNode, fieldType, self.engine.getService(), path, operationType);
+        Field 'field = getFieldObject(fieldNode, operationType, self.schema, self.engine);
         self.context.resetInterceptorCount();
         var result = self.engine.resolve(self.context, 'field);
         self.errors = self.context.getErrors();
         self.data[fieldNode.getAlias()] = result is ErrorDetail ? () : result;
```

Comment:
Seems these 4 lines repeated in all three methods, we could move this to a function